auth/proxy/jwk-cache.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers */
+import { JwksFetchError } from "./errors";
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const JWKS_URI = 'https://firebaseappcheck.googleapis.com/v1/jwks';
+
+interface Key {
+    kid: string;
+    nbf?: number;
+    use: string;
+    kty: string;
+    e: string;
+    n: string;
+}
+
+interface JwksCache {
+    jwks: Jwks;
+    expiresAt: number
+}
+
+const cacheDuration = 5; // Caching for 5 minutes
+
+const isJwks = (responseJson: unknown): responseJson is Jwks => {
+    return (
+        typeof responseJson === 'object' &&
+        responseJson !== null &&
+        Array.isArray((responseJson as { keys?: unknown }).keys)
+    );
+};
+
+let cachedJwks: JwksCache | null = null;
+
+interface Jwks {
+    keys: Key[];
+}
+
```

Comment:
Add a parser for the cache-control header [see](https://github.com/govuk-one-login/mobile-platform-back/blob/main/backend-api/src/request/parseCacheControlHeader.ts#L1)